### PR TITLE
add tests and refactor code to filter image attachments

### PR DESCRIPTION
**Pull request by lucatume**
Ticket (context): https://central.tri.be/issues/96393

This PR:

* refactors EA Client code to allow filtering the image attachments for events, venues and organizers
* refactors the code
* adds tests

### CODE REVIEW

**Comment by tr1b0t** [comment]
Closing this Pull Request because it is pull requested against the wrong branch. The Central ticket is for **, but this was pull requested against *release/M18.01*.

File: src/Tribe/Aggregator/Record/Abstract.php [hunk 1]
```diff
@@ -2072,18 +2044,24 @@ public function import_aggregator_image( $event ) {
...
+	 * @param array $data A post data in array format.
 	 *
 	 * @return object|bool An object with the image post ID or `false` on failure.
 	 */
-	public function import_image( $event ) {
-		if ( empty( $event['image'] ) || ! filter_var( $event['image'], FILTER_VALIDATE_URL ) ) {
+	public function import_image( $data ) {
+		if (
+			empty( $data['image'] )
+			&& (
```
Comment (barryhughes) [inline src/Tribe/Aggregator/Record/Abstract.php]: :thinking: ... if `$data['image']` is empty surely the result of the next part of the conditional will also always be true? Is the second part needed, or else should we be doing something other than `&&` here?

File: src/Tribe/Aggregator/Record/Abstract.php [hunk 2]
```diff
@@ -2253,21 +2231,87 @@ public function should_queue_import( $should_queue_import = null ) {
...
+	 * @return bool Whether the image was attached to the venue or not.
+	 */
+	public function import_venue_image( $venue_id, $image_url, $activity ) {
+		/**
+		 * Whether the venue image should be imported and attached or not.
+		 *
+		 * @since TBD
+		 *
+		 * @param bool                                        $import_venue_image Defaults to `true`
+		 * @param int                                         $venue_id       The venue post ID
```
Comment (barryhughes) [inline src/Tribe/Aggregator/Record/Abstract.php]: Doesn't matter too much but the spacing/alignment is somewhat out of whack after `$venue_id` and before the description

File: src/Tribe/Aggregator/Record/Abstract.php (hunk 1 above)
Comment (lucatume) [reply src/Tribe/Aggregator/Record/Abstract.php]: You are correct! I fixed this and pushed.

File: src/Tribe/Aggregator/Record/Abstract.php (hunk 2 above)
Comment (lucatume) [reply src/Tribe/Aggregator/Record/Abstract.php]: Correct again, was Chaotic Evil, I've corrected to Lawful Good.

**Comment by lucatume** [comment]
This is ready for another look (and can be merged in my absence if need be).

**Pull request merged**